tensorboard/plugins/projector/vz_projector/data.ts

```diff
@@ -370,6 +366,92 @@ export class DataSet {
     });
   }
 
+  /** Runs UMAP on the data. */
+  async projectUmap(nComponents: number, nNeighbors: number, stepCallback: (iter: number) => void) {
+    this.hasUmapRun = true;
+    this.umap = new UMAP({nComponents, nNeighbors});
+
+    let nEpochs = 1;
+    let currentEpoch = 0;
+    const epochStepSize = 10;
+    const sampledIndices = this.shuffledDataIndices.slice(0, UMAP_SAMPLE_SIZE);
+  
+    const sampledData = sampledIndices.map(i => this.points[i]);
+    // TODO: Switch to a Float32-based UMAP internal
+    const X = sampledData.map(x => Array.from(x.vector));
+
+    this.nearest = await this.computeKnn(sampledData, nNeighbors);
+    
+    nEpochs = await util.runAsyncTask('Initializing UMAP...', () => {
```

Comment:
The initialization takes a while, maybe 10-20s for the demo dataset (n=5000, d=784)... There are a few steps being performed in this phase, and I've thought about later updating umap-js to expose the progress in these steps via a callback to display amore helpful message in the UI, but I thought this .

The `initializeFit` function is synchronous, and the promise returned by `util.runAsyncTask` is resolved after it is completed. I think that this variant of the helper function was originally created to "buffer" computationally expensive synchronous actions (like updating the positions of points in the t-SNE optimization steps) with some timeout time around them, in order to prevent the UX from freezing up during these steps. 

Are you saying I should use the more generic `tf_graph_common/util.ts` helper methods here? The only bummer that I see with that is the embedding projector doesn't conform to the `ProgressTracker` API in the generic util.